harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/geonet/Aligner.java

```diff
@@ -883,12 +883,23 @@ private void saveFile(final Store store, String metadataUuid, String file,
         ISODate remIsoDate = new ISODate(changeDate);
         boolean saveFile;
 
-        final MetadataResource description = store.getResourceDescription(context, metadataUuid, visibility, file, true);
-        if (description == null) {
+        Store.ResourceHolder resourceHolder;
+        try {
+            resourceHolder = store.getResource(context, metadataUuid, visibility, file, true);
+        } catch (ResourceNotFoundException ex) {
+            resourceHolder = null;
+        }
+
+        if (resourceHolder == null) {
```

Comment:
Would the following be better?
`if (resourceHolder == null || resourceHolder.getMetadata() == null) {`

Then yo would not need 

```
if (description == null) {
                saveFile = true;
            } else {
```

Reply:
Indeed @ianwallen, I'll update the code. Thanks.

Reply:
@ianwallen updated the pull request, used `&&` to check for not null, otherwise the check you proposed can cause a `NullPointerException`in `resourceHolder.getMetadata()` when `resourceHolder` is  `null`.

Reply:
I don't understand how you can get a NPE with the following

`if (resourceHolder == null || resourceHolder.getMetadata() == null) {`

because if `resourceHolder == null` is true it will not evaluate the next condition.  So `resourceHolder.getMetadata() == null` will only be evaluated if `resourceHolder != null`

In either case, your new changes looks good

Thank you

Reply:
@ianwallen you are right, I mixed up the check.